test/test-checkout.sh

```diff
@@ -155,3 +155,94 @@ begin_test "checkout: outside git repository"
   grep "Not in a git repository" checkout.log
 )
 end_test
+
+begin_test "checkout: with include"
+(
+  set -e
+
+  reponame="checkout-with-include"
+  setup_remote_repo "$reponame"
+
+  clone_repo "$reponame" repo2
+
+  git lfs track "*.dat" 2>&1 | tee track.log
+  grep "Tracking \"\*.dat\"" track.log
+
+  contents="something something"
+  contentsize=19
+  contents_oid=$(calc_oid "$contents")
+
+  # Same content everywhere is ok, just one object in lfs db
+  printf "$contents" > file1.dat
+  printf "$contents" > file2.dat
+  printf "$contents" > file3.dat
+  mkdir folder1 folder2
+  printf "$contents" > folder1/nested.dat
+  printf "$contents" > folder2/nested.dat
+  git add file1.dat file2.dat file3.dat folder1/nested.dat folder2/nested.dat
+  git add .gitattributes
+  git commit -m "add files"
+
+  [ "$contents" = "$(cat file1.dat)" ]
+  [ "$contents" = "$(cat file2.dat)" ]
+  [ "$contents" = "$(cat file3.dat)" ]
+  [ "$contents" = "$(cat folder1/nested.dat)" ]
+  [ "$contents" = "$(cat folder2/nested.dat)" ]
+
+  # Remove the working directory
+  rm -rf file1.dat file2.dat file3.dat folder1/nested.dat folder2/nested.dat
```

Comment:
I'm having a little bit of trouble with this line. The normal behavior for `git lfs checkout` is to smudge pointers in the working copy to turn them into large objects. While `git {,lfs} checkout` will create files if they don't exist, I don't think that this is the approach we should use for testing.

Instead, what about something like this:

```sh
begin_test "checkout: setup for --include --exclude"
(
  set -e
  reponame="checkout-include-exclude"
  setup_remote_repo "$reponame"
  clone_repo "$reponame"

  # Initialize exactly as L168-184.
)
end_test

begin_test "checkout: with include"
(
  set -e

  cd "$TRASHDIR"
  # Manually clone without downloading any objects.
  git -c "lfs.fetchexclude=*" clone "$GITSERVER/checkout-include-exclude"
  cd checkout-include-exclude

  git config credential.helper lfstest

  # Test as in line L196-197
)
end_test
```

This has the benefit of moving repeated initialization into one test, and shortening the behavior tests.

Reply:
Akaun